Windows specific dead code cleanup

SudeshnaMoulik opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
We have our server/sched/comm components for Linux specific. This PR is for removal of Windows specific code for those corresponding modules. 


#### Describe Your Change
Clean Windows dead code for below modules:
server (along with database)
sched
comm
tcl/tk
send_hooks
send_job
tools
wix/prep.bat cleanup


#### Link to Design Doc
<!--- If there is a design, link to it here: **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)** -->


#### Attach Test and Valgrind Logs/Output
Regression test (perl and ptl) is successful (job id: 63)
Build and Test across platform is successful (20.0.0.20191119101323 and 20.0.0.20191119041123)

Manual Windows test logs (34 tests) : 
[Win_Lin_testlogs.txt](https://github.com/PBSPro/pbspro/files/3878800/Win_Lin_testlogs.txt)




<!--- Pull Request Guidelines: [Pull Request Guidelines](https://pbspro.atlassian.net/wiki/spaces/DG/pages/1187348483/Pull+Request+Guidelines) -->

bhroam commented on src/server/svr_movejob.c:
```diff
@@ -702,200 +672,6 @@ int
-	win_pwrite(&pio, buf, strlen(buf));
-	win_pclose2(&pio);	/* closes all handles except the process handle */
-	return (2);
-#else
```
Shouldn't there be a removal of an endif that goes with the removal of this else?

SudeshnaMoulik replied:
Yes @bhroam, that is removed at the end of send_job function:
https://github.com/PBSPro/pbspro/pull/1414/files/efa51fb0c8837da4d6c3953a7895138f83ee28d3#diff-a51b4b3581fd3b2b0856a1ae6763b1f4L1269

bhroam commented on src/tools/pbs_ds_monitor.c:
```diff
@@ -183,25 +162,18 @@ get_pid()
- *		lockfile.
+ *		For Unix, this uses fcntl lock (not inheritable).
+ *      If the operand is F_WRLCK, then this also writes
+ *      the pid of this process to the lockfile.
```
Is the indentation off here?

SudeshnaMoulik commented on src/server/pbsd_main.c:
```diff
@@ -1734,25 +1442,16 @@ main(int argc, char **argv)
-#endif	/* end the ifndef WIN32 */
 	pbs_close_stdfiles();
 #else	/* DEBUG is defined */
-#ifndef WIN32
```
Thanks @bhroam for pointing this out.

subhasisb commented on src/server/pbsd_init.c:
```diff
@@ -478,11 +455,8 @@ pbsd_init(int type)
 	}
-#endif	/* WIN32 */
 
 #else	/* setrlimit 32 bit */
```
I believe we do not need the 32 bit limit setting code, since server (pbsd_init.c is part of server only), is built only in 64 bit mode on all platforms now that windows server is gone

subhasisb commented on src/server/pbsd_init.c:
```diff
@@ -673,9 +617,6 @@ pbsd_init(int type)
-#ifdef WIN32
-		if (stalone == 1)
-#endif
 			need_y_response(type, "no server database exists");
```
indent one left

subhasisb commented on src/server/pbsd_init.c:
```diff
@@ -771,9 +712,6 @@ pbsd_init(int type)
-#ifdef WIN32
-			if (stalone == 1)
-#endif
 				need_y_response(type, "server database exists");
```
indent one left

subhasisb commented on src/server/pbsd_main.c:
```diff
@@ -506,10 +476,6 @@ rpp_request(int fd)
-			/* workaround to a win2k winsock bug */
-			if (errno != 10054)
-#endif
 				log_err(errno, __func__, "rpp_poll");
```
indent one left

subhasisb commented on src/server/svr_chk_owner.c:
```diff
@@ -43,12 +43,14 @@
- * 	svr_force_disconnect()
- *
+ * 	svr_chk_owner()
+ *  svr_authorize_jobreq()
```
indent

subhasisb commented on src/server/svr_migrate_data.c:
```diff
@@ -64,14 +64,6 @@
 #include <sys/stat.h>
 #include <libutil.h>
 
```
Remove this file

SudeshnaMoulik commented:
Thanks @subhasisb for the review.
I have addressed all your comments, except removal of svr_migrate_data.c file. As server uses one function from it [svr_migrate_data() - in case for PBS version change there are any server data structure change happened). Although removed other functions from it. Please review the same.

SudeshnaMoulik commented:
Re-based branch and squashed multiple commits.

SudeshnaMoulik commented:
Regression across all platform is successful.
